### PR TITLE
fix: BSX Assets page broken

### LINKED ISSUE #8023: BSX Assets page broken
image

1. Balance is 0 
2. I have just set the assets token to be KSM =< refresh => shows bsx again

### Related:

I don't think that is the right deposit amount. it is probably should be BSX , not KSM (too expansive)
image

### PR DESCRIPTION
**Pull request by hassnian**
👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #8023

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2023-11-12 at 10 17 56@2x

### <samp>🤖 Generated by Copilot at e37d205</samp>

Refactored and improved the data fetching and rendering logic for the `AssetList` and `AssetTable` components in `bsx/Asset`. Added a loading indicator for the `AssetTable`.

### <samp>🤖 Generated by Copilot at e37d205</samp>

> _We are the masters of the asset list_
> _We use a local ref to resist_
> _The mutation of the prop that we despise_
> _We show the loading state until the data arrives_

### CODE REVIEW

**Review by roiLeo — APPROVED** [review]
✅ code lgtm & wfm

**Review by preschian — APPROVED** [review]

**Comment by prury** [comment]
Firefox problem: i can't click on any button, they are all deactivated
image

**Comment by prury** [comment]
> > Firefox problem: i can't click on any button, they are all deactivated
> 
> I'm using firefox too but the page works well, did you clear your browser cache? which chain are you testing?

i did clear my cache, bsx is the only one that have the assets page

**Comment by prury** [comment]
> CleanShot 2023-11-13 at 17 38 48@2x maybe bsx down ?

works fine on chrome, but it seems to me that is a loading problem:

https://github.com/kodadot/nft-gallery/assets/36627808/3be80fc7-d436-4a8f-910b-c63cc2aecdb3

**Comment by hassnian** [comment]
> Edit: yes, it seems that bsx has stopped now, but there's still this strange background color when loading

that's the loading state of the table component , 

 I thought I was by design but I guess is not, should I add a spinner or this state is intended  ? @exezbcz

**Comment by exezbcz** [comment]
> I thought I was by design

nope, no design was done on this page yet. 




> I add a spinner

ideally yes for now

**Comment by hassnian** [comment]
CleanShot 2023-11-16 at 11 08 24

**Comment by hassnian** [comment]
@roiLeo added https://github.com/kodadot/nft-gallery/pull/8024/commits/015e5db25fcdc03471e9499aa10357419ce5c688 pls check if any comments

File: libs/ui/src/components/NeoLoading/NeoLoading.vue [hunk 1]
```diff
@@ -1,8 +1,16 @@
+<template>
+  <OLoading v-bind="$attrs">
+    <slot>
+      <NeoIcon icon="spinner-third" spin size="large" class="has-text-white" />
+    </slot>
+  </OLoading>
+</template>
```
Comment (roiLeo) [inline libs/ui/src/components/NeoLoading/NeoLoading.vue:7]: this component is used in 2 loaders, will this affect the design?

File: libs/ui/src/components/NeoLoading/NeoLoading.vue [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
 <script lang="ts">
 import { OLoading } from '@oruga-ui/oruga-next'
+import { NeoIcon } from '../..'
```
Comment (roiLeo) [inline libs/ui/src/components/NeoLoading/NeoLoading.vue:10]: hmmm weird import

File: libs/ui/src/components/NeoTable/NeoTable.vue [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
+
+    <template #loading>
+      <NeoLoading :full-page="false" :active="$attrs.loading" />
+    </template>
+  </OTable>
+</template>
+
 <script lang="ts">
 import { OTable } from '@oruga-ui/oruga-next'
+import { NeoLoading } from '../..'
```
Comment (roiLeo) [inline libs/ui/src/components/NeoTable/NeoTable.vue:16]: import

File: libs/ui/src/components/NeoTable/NeoTable.vue [hunk 4]
```diff
@@ -1,8 +1,22 @@
+<template>
+  <OTable v-bind="$attrs">
+    <!-- Pass on all named slots -->
+    <template v-for="(_, slot) in $slots" #[slot]="scope">
+      <slot :name="slot" v-bind="scope || {}" />
+    </template>
+
+    <template #loading>
+      <NeoLoading :full-page="false" :active="$attrs.loading" />
```
Comment (roiLeo) [inline libs/ui/src/components/NeoTable/NeoTable.vue:9]: or you can put dirrectly your new code here

**Review by roiLeo — COMMENTED** [review]
will it work with our current Loaders?

File: libs/ui/src/components/NeoLoading/NeoLoading.vue (hunk 1 above)
Comment (hassnian) [reply libs/ui/src/components/NeoLoading/NeoLoading.vue:7]: nope, it only replaces the default loader slot 

Loader.vue : normal 
CleanShot 2023-11-16 at 13 33 37@2x
loader.vue with not slot
CleanShot 2023-11-16 at 13 34 10@2x
CleanShot 2023-11-16 at 13 34 26@2x

**Comment by yangwao** [comment]
BSX is being deprecated currently so please don't invest much effort to it. 

Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 5.53 USD/DOT ~ 5.425 $DOT
🧗 [13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh](https://polkadot.subscan.io/account/13QUj3pZyFNfYj4AM336hRdyLQbevj5H3sR4PKmLEXLdwZhh)
🔗 [0xa3310e21f70dfd93e4fd6f387d56054415fba78dbc2606d893aefa66103a2a7d](https://polkadot.subscan.io/extrinsic/0xa3310e21f70dfd93e4fd6f387d56054415fba78dbc2606d893aefa66103a2a7d)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**